internal/scheduler/scheduler.go

```diff
@@ -356,6 +356,54 @@ func (s *Scheduler) cycle(ctx *armadacontext.Context, updateAll bool, leaderToke
 	return overallSchedulerResult, nil
 }
 
+// syncInitialState updates jobs in jobDb to match state in postgres and returns all in-progress jobs.
+func (s *Scheduler) syncInitialState(ctx *armadacontext.Context) ([]*jobdb.Job, []jobdb.JobStateTransitions, error) {
```

Comment:
I *think*  the only difference between this function and `syncState` needs to be the queries we use.  E.g. `s.jobRepository.FetchInitialJobs`  vs `s.jobRepository.FetchJobs`.  We could just modify syncState to take an `initialise` parameter of type bool which selects the appropriate query.

Reply:
Looks good, but I think there's in issue with potentially missed updates in `syncInitialState`.  I think we can fix this by simply modifying `syncState` to use the new queries rather than creating a separate  ` syncInitialState` function but happy to entertain other solutions.

Reply:
I tI think an edge case is that the serials could still be quite old but in most cases they should be fairly up to date so I'm ok with simplifying things.